pkg/daemon/update.go

```diff
@@ -732,7 +979,48 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig, skipCertifi
 	logSystem("Starting update from %s to %s: %+v", oldConfigName, newConfigName, diff)
 
 	diffFileSet := ctrlcommon.CalculateConfigFileDiffs(&oldIgnConfig, &newIgnConfig)
-	actions, err := calculatePostConfigChangeAction(diff, diffFileSet)
+	diffUnitSet := ctrlcommon.CalculateConfigUnitDiffs(&oldIgnConfig, &newIgnConfig)
+
+	var fg featuregates.FeatureGate
+
+	// This check is needed as featureGatesAccessor is not present during first boot. During firstboot
+	// the daemon will always do a reboot and NodeDisruptionPolicies are not active.
+	if dn.featureGatesAccessor != nil {
+		fg, err = dn.featureGatesAccessor.CurrentFeatureGates()
+		if err != nil {
+			klog.Errorf("Could not get fg: %v", err)
+			return err
+		}
+	}
+
+	var nodeDisruptionActions []opv1.NodeDisruptionPolicyStatusAction
+	var actions []string
+	// If FeatureGateNodeDisruptionPolicy is set, calculate NodeDisruptionPolicy based actions for this MC diff
+	if fg != nil && fg.Enabled(v1.FeatureGateNodeDisruptionPolicy) {
+		mcop, err := dn.mcopClient.OperatorV1().MachineConfigurations().Get(context.TODO(), "cluster", metav1.GetOptions{})
```

Comment:
I realized in https://github.com/openshift/machine-config-operator/pull/4281/  you added this as "MCOOperatorKnobsObjectName". We should eventually merge the two :)

Reply:
ah good catch, I'll make sure to pick it up on next pass!